src/mame/includes/ladybug.h

```diff
@@ -84,52 +85,80 @@ class dorodon_state : public ladybug_state
 	required_shared_ptr<uint8_t> m_decrypted_opcodes;
 };
 
-
-// graphics from ladybug, stars from zerohour, plus grid layer
-class sraider_state : public ladybug_base_state
+// 8203 hardware adds grid tilemap, 2nd audio CPU
+class mrsdyna_state : public ladybug_base_state
 {
 public:
-	sraider_state(const machine_config &mconfig, device_type type, const char *tag)
+	mrsdyna_state(const machine_config &mconfig, device_type type, const char *tag)
 		: ladybug_base_state(mconfig, type, tag)
-		, m_grid_data(*this, "grid_data")
 		, m_palette(*this, "palette")
 		, m_gfxdecode(*this, "gfxdecode")
 		, m_video(*this, "video")
-		, m_stars(*this, "stars")
+		, m_soundlatch_low(*this, "soundlatch_low")
+		, m_soundlatch_high(*this, "soundlatch_high")
 	{ }
 
-	void sraider(machine_config &config);
+	void mrsdyna(machine_config &config);
 
 protected:
-	uint8_t sraider_8005_r();
-	void sraider_misc_w(offs_t offset, uint8_t data);
-	void sraider_io_w(uint8_t data);
-	void sraider_palette(palette_device &palette) const;
+	uint8_t mrsdyna_protection_r();
+	void mrsdyna_weird_w(offs_t offset, uint8_t data);
+	void mrsdyna_0x30_w(offs_t offset, uint8_t data);
+	void mrsdyna_0x38_w(offs_t offset, uint8_t data);
+	uint8_t mrsdyna_rnd_r();
+	void mrsdyna_io_w(uint8_t data);
+	void mrsdyna_palette(palette_device &palette) const;
 	DECLARE_WRITE_LINE_MEMBER(screen_vblank_sraider);
 	TILE_GET_INFO_MEMBER(get_grid_tile_info);
 
 	virtual void machine_start() override;
 	virtual void machine_reset() override;
-	virtual void video_start() override;
-	uint32_t screen_update_sraider(screen_device &screen, bitmap_ind16 &bitmap, const rectangle &cliprect);
+	uint32_t screen_update_mrsdyna(screen_device &screen, bitmap_ind16 &bitmap, const rectangle &cliprect);
 
-	void sraider_cpu1_map(address_map &map);
-	void sraider_cpu2_io_map(address_map &map);
-	void sraider_cpu2_map(address_map &map);
+	void mrsdyna_cpu1_map(address_map &map);
+	void mrsdyna_cpu2_io_map(address_map &map);
+	void mrsdyna_cpu2_map(address_map &map);
 
-private:
-	required_shared_ptr<uint8_t> m_grid_data;
 	required_device<palette_device> m_palette;
 	required_device<gfxdecode_device> m_gfxdecode;
 	required_device<ladybug_video_device> m_video;
-	required_device<zerohour_stars_device> m_stars;
+	required_device<latch8_device> m_soundlatch_low;
+	required_device<latch8_device> m_soundlatch_high;
```

Comment:
There’s no point having these here if you don’t actually use them in the machine configuration and address map.  Either use the object finders in place of explicit tags in the address maps and machine configuration, or don’t add these object finder members (if you choose the latter, you can also punt the `#include "machine/latch8.h"` over to the .cpp file).

Reply:
I went over this in detail with @smf- and I understand how it works now.  I chose the latter.